### PR TITLE
Pager and PagerSlim shapes

### LINKED ISSUE #5461: Update Pager and PagerSlim shapes documentation
Update Pager and PagerSlim shapes documentation

### PR DESCRIPTION
**Pull request by Skrypt**
TODO : 

* [x] Assert that Attributes work when using Pager
* [x] Update Pager Documentation

Closes #5461

### CODE REVIEW

**Comment by Skrypt** [comment]
Here what I found is that the Shape class already declares a _attributes private field and also an Attributes property which is initialized with an empty Dictionary. Maybe that's why it doesn't take in consideration what we are passing when using an Attributes param on a Shape.

**Comment by Skrypt** [comment]
Ok I'm done on this one. I think it works fine with my changes. I removed a lot of dynamic Shape declarations because I didn't like it.

**Comment by Skrypt** [comment]
@sebastienros Ok what we found last night is that when we create a pager for example on a controller.

```
var pagerShape = (await New.Pager(pager)).TotalItemCount(maxPagedCount > 0 ? maxPagedCount : await query.CountAsync()).RouteData(routeData).TagName("div").ItemTagName("div").Id("someid");
```

In the debugger it will affect Shape.TagName and Shape.ItemTagName ... 
And it will also affect Shape.Properties["TagName"] ...
But if we do :   

```
var test = Shape.TagName;
```

Right after, the value of `test` will be equal to Null.
So, here what we looked at last night is that when a method is marked with the [Shape] attribute it will require to pass a param of the same name than what is affected previously for it to get bind by the ShapeBindingStrategy class. But, when a property of the Shape class is passed it is not required.

Though, after changing `public async Task<IHtmlContent> Pager_Links(dynamic Shape,` to `public async Task<IHtmlContent> Pager_Links(Shape Shape,` I thought that I would be able to do Shape.TagName after adding this as a public property of the Shape class but it always returns Null because of the way that we create and assign these parameters when building a `var pagerShape = (await New.Pager(pager))`. If we would do pagerShape.TagName = "div"; after creating the shape it would affect the value to the class instance else it only affect the Shape.Properties["TagName"] value. So, since we can't rely on someone assigning these values to the Shape class properties and that we want to create these shapes in one line of code I used Shape.Properties["something"] everywhere where we we're using dynamic Shape to access these dynamic properties. I removed dynamic Shape everywhere that I could to optimize this code perf as a result.

File: src/OrchardCore/OrchardCore.DisplayManagement/Shapes/CoreShapes.cs [hunk 1]
```diff
@@ -49,22 +43,29 @@ public void PlaceChildContent(dynamic Source, TextWriter Output)
                 return HtmlString.Empty;
             }
 
-            string listTagName = null;
+            var listTagName = "ul";
 
-            if (Tag != "-")
+            if (Shape.Properties.TryGetValue("TagName", out var value))
```
Comment (jtkech) [inline src/OrchardCore/OrchardCore.DisplayManagement/Shapes/CoreShapes.cs]: As in other places, better to use `(...&& value is string valueString)` in place of using `ToString()` below.

File: src/OrchardCore/OrchardCore.DisplayManagement/Shapes/CoreShapes.cs [hunk 2]
```diff
@@ -49,22 +43,29 @@ public void PlaceChildContent(dynamic Source, TextWriter Output)
...
-                listTagName = string.IsNullOrEmpty(Tag) ? "ul" : Tag;
+                if (!String.IsNullOrEmpty(value.ToString()) && value.ToString() != "-")
+                {
+                    listTagName = value.ToString();
+                }
             }
 
-            var listTag = String.IsNullOrEmpty(listTagName) ? null : Shape.GetTagBuilder(listTagName, Id, Classes, Attributes);
+            string id = null;
+            if (Shape.Properties.TryGetValue("Id", out var IdValue)) {
```
Comment (jtkech) [inline src/OrchardCore/OrchardCore.DisplayManagement/Shapes/CoreShapes.cs]: idem

and formatting, brace at the new line

**Comment by jtkech** [comment]
As a reminder:

- Okay if the goal is to replace `dynamic shape` with `Shape shape`.

- Okay with the new and real `TagName` `IShape` property, allowing, as for the `Id` property, to init a `Shape` type instance with these props, and knowing that in fact they can't be used as real props when we create a dynamic shape by just passing dynamic named parameters, as we do currently in some places, in that case we need to use e.g `shape.Properties["TagName"]`.

**Comment by Skrypt** [comment]
I merged and I'm not stressed at all with this one. It fixes the issues it needs tp fix. There's plenty of ways to do this but I think removing the dynamic Shapes and using that Properties dictionary should be quicker in the end. I could have left it like it was but ... I was working on it and I followed what @sebastienros said. If I made a mistake or I did not understand properly I don't get it at this point. The only thing that I did not get totally is the ShapeBindingStrategy and why `Shape.TagName` for example is not affected when doing` (await New.Pager(pager))`. Maybe the next step would be to actually fix that part to bind it. But I've not looked at this part. The goal of the PR was mainly to be able to affect TagName, Attributes, which we're missing in the implementation.

**Pull request merged**